File: 2026-01/src/day_15/boj_2920/Main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);

    vector<int> v(8);
    for (int& x : v) cin >> x;

    bool isAsc = true;
    bool isDesc = true;

    for (int i = 1; i < 8; i++) {
        if (v[i] < v[i - 1]) {
            isAsc = false;
        }

        if (v[i] > v[i - 1]) {
            isDesc = false;
        }
    }

    if (isAsc) {
        cout << "ascending";
    } else if (isDesc) {
        cout << "descending";
    } else {
        cout << "mixed";
    }
}

File: 2026-01/src/day_15/boj_2920/Main.java
package day_15.boj_2920;

import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int[] arr = new int[8];
        for (int i = 0; i < 8; i++) {
            arr[i] = Integer.parseInt(st.nextToken());
        }

        boolean isAsc = true;
        boolean isDesc = true;

        for (int i = 1; i < 8; i++) {
            if (arr[i] < arr[i - 1]) {
                isAsc = false;
            }

            if (arr[i] > arr[i - 1]) {
                isDesc = false;
            }
        }

        if (isAsc) {
            System.out.println("ascending");
        } else if (isDesc) {
            System.out.println("descending");
        } else {
            System.out.println("mixed");
        }
    }
}
